### DISCUSSION TITLE
How can we support agent/agentless monitoring and active/passive modes?

### DISCUSSION THREAD

**Discussion by tomorrowshipyltm** [apache/hertzbeat#2178] [Q&A]
Hello everyone, previously our team has used Zabbix. I would like to know how to achieve the following features based on HertzBeat:

1. Support for both agent and agentless monitoring.
2. Agent-based monitoring with active and passive modes. In active mode, the agent requests and receives monitoring configurations from the server, then periodically collects and sends monitoring data to the server. For monitoring thousands of machines, it is recommended to use the active mode to avoid overwhelming the server. In passive mode, the agent receives data collection requests from the server, collects the data, and sends it back to the server.
3. Agentless monitoring, where data is collected directly from the target host using standard protocols such as SSH, HTTP, ICMP, etc.
4. Flexible deployment with the use of proxy intermediaries based on network environment and cluster scale. For example, Zabbix's server-proxy-client architecture is suitable for medium-sized network architectures spanning multiple data centers or networks. The proxy acts as a communication bridge between the server and client. It does not store data but temporarily holds the data received from agents before submitting it to the server.

My understanding of the above issues:

1. If push module is used to support agents, then:
     (1)The communication protocol and command interaction between the server and agent can be complex. The push module might include many functions, such as health check and data request components.
    (2)Active mode is not supported.
4. For flexible deployment, is it possible to use multiple collectors?

**Comment by tomsun28** [comment]
hi, sorry for slow response. The `push module` is developing, we will support it int the future. 
> point 1

for agent monitoring, now you can deploy a collector in your peer machine, and use it as a agent.  But the collector is not a regular agent.
> point 2 

now our collector mode is sending collect data to main server. 
> point 3 

yes.
> point 4

we support deploy multiple collectors in different private network, and the collector communicate with main server.
